ubuntu-deploy/index.md

```diff
@@ -0,0 +1,414 @@
+author:
+  name: 'Yusuf Akorede'
+  avatar_file_name: 'avatar.png'
+---
+
+## Introduction
+
+In this tutorial, you will learn how to deploy a Directus application within a Docker container on an Ubuntu Linux server and connect it to a custom domain, such as `https://directus.exampledomain.com`.
+
+This guide covers setting up Docker, configuring Docker Compose, using Nginx as a reverse proxy, and securing your application with SSL. Additionally, you'll discover how to run the application as a background service, ensuring seamless operation and easy management.
+
+## Prerequisites
+
+1. **A Directus Project:** Prepare a local Directus project for deployment. Follow the [Directus quickstart guide][quickstart] if you need to create a new project.
+
+2. **Ubuntu Server (Version 20.04 or 22.04):** Access an Ubuntu server via SSH. Obtain one from cloud providers like Azure, DigitalOcean, Linode, or AWS. Configure SSH access from your local machine.
+
+3. **Domain Name:** Register a domain name and have access to its DNS settings. This is crucial for the domain to resolve to your Ubuntu server
+
+4. **Basic Command-Line Skills:** Basic familiarity with Linux command-line operations, including file uploads using `scp` and editing files with `vi`.
+
+## Upload Your Local Directus Application Folder to the Server
+
+Assuming you have a local Directus folder structure that looks like this:
+
+```plaintext
+├── database
+│   └── data.db
+├── docker-compose.yml
+├── uploads
+└── extensions
+```
+
+And the _docker-compose.yml_ file looks somewhat like this:
+
+```yaml
+version: '3'
+services:
+  directus:
+    image: directus/directus:latest
+    ports:
+      - 8055:8055
+    volumes:
+      - ./database:/directus/database
+      - ./uploads:/directus/uploads
+      - ./extensions:/directus/extensions
+    environment:
+      KEY: 'replace-with-random-value'
+      SECRET: 'replace-with-random-value'
+      ADMIN_EMAIL: 'admin@example.com'
+      ADMIN_PASSWORD: 'd1r3ctu5'
+      DB_CLIENT: 'sqlite3'
+      DB_FILENAME: '/directus/database/data.db'
+      WEBSOCKETS_ENABLED: 1
+      # Add your other settings
+```
+
+:::info Security Note
+
+Ensure you adjust the email and set a strong password for the admin user.
+
+:::
+
+Use `scp` (Secure Copy Protocol) to upload the local folder to your server.
+
+From your local machine terminal, run:
+
+```bash
+scp -r /path/to/your/local/directus/folder username@server_ip:/path/to/your/remote/folder
+```
+
+Replace _/path/to/your/local/directus/folder_ with the actual local path of your Directus application directory, and _/path/to/your/remote/folder_ with the respective path on your server.
+
+Also, replace `username` with your server's username and `server_ip` with your server's public IP address.
+
+In the example below, I am copying the whole Directus folder to the home directory(`~`) of my server.
+
+![Copying files to the server][image-6]
```

Comment:
Can you briefly explain why only the data.db and compose file are shown in the terminal here?

Reply:
I think it is to improve readability and avoid cluttering the output. One can easily identify the files being transferred without displaying the entire directories